File: Civitas-Ruby/lib/casilla_impuesto.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative "diario"
require_relative "mazo_sorpresas"
require_relative "titulo_propiedad"
require_relative "sorpresa"
require_relative "tipo_casilla"
require_relative "tipo_sorpresa"
require_relative "jugador"

module Civitas
  class CasillaImpuesto < Casilla

    def initialize(cantidad, nombre)
      super(nombre)
      @importe = cantidad
    end
    
    @Override
    def recibejugador(actual, todos)
      if jugadorcorrecto(actual, todos)
        informe(actual, todos)
        todos[actual].pagaimpuesto(@importe)
      end
    end
    
    @Override
    def to_s
      '\nNombre: ' + super.nombre.to_s +
      '\nTipo Casilla: IMPUESTO' + 
      '\nImporte: ' + @importe.to_s
    end
  end
end

File: Civitas-Java/Civitas/src/main/java/civitas/CasillaImpuesto.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 * @author elena
 * @date 29/11/2019
 */
public class CasillaImpuesto extends Casilla{
    private float importe = 0;
    
    CasillaImpuesto(float cantidad, String nombre){ 
        super(nombre);
        this.importe = cantidad;
    }
    @Override
    public boolean soyCasillaCalle(){ return false; }
    @Override
    void recibeJugador(int iactual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(iactual, todos)){
            super.informe(iactual, todos);
            todos.get(iactual).pagaImpuesto(this.importe);
        }
    }
    
    @Override
    public String toString(){
        return "\nNombre: " + super.getNombre() +
               "\nTipo Casilla: IMPUESTO" +
               "\nImporte: " + this.importe;
    }
}
